Formula/cmctl.rb

```diff
@@ -0,0 +1,27 @@
+class Cmctl < Formula
+  desc "Command-line tool to manage cert-manager"
+  homepage "https://cert-manager.io"
+  url "https://github.com/cert-manager/cert-manager.git",
+    tag:      "v1.10.0",
+    revision: "da3265115bfd8be5780801cc6105fa857ef71965"
+  license "Apache-2.0"
+  head "https://github.com/cert-manager/cert-manager.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/cert-manager/cert-manager/cmd/ctl/pkg/build.name=cmctl
+      -X github.com/cert-manager/cert-manager/cmd/ctl/pkg/build/commands.registerCompletion=true
+      -X github.com/cert-manager/cert-manager/pkg/util.AppVersion=v#{version}
+      -X github.com/cert-manager/cert-manager/pkg/util.AppGitCommit=#{Utils.git_head}
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags.join(" ")), "./cmd/ctl"
+    generate_completions_from_executable(bin/"cmctl", "completion")
+  end
+
+  test do
+    assert_match version.to_s, shell_output("cmctl version --client")
```

Comment:
👋 Hey! Thank you for the review comments.

The main barrier is that cmctl requires a Kubernetes cluster with cert-manager installed for most of its functinality to work (similar to other tools in homebrew-core such as [istioctl](https://github.com/Homebrew/homebrew-core/blob/5b36ba740a61cab0a4f3b83c269baa05a3b0c538/Formula/istioctl.rb), [k9s](https://github.com/Homebrew/homebrew-core/blob/5b36ba740a61cab0a4f3b83c269baa05a3b0c538/Formula/k9s.rb), [velero](https://github.com/Homebrew/homebrew-core/blob/5b36ba740a61cab0a4f3b83c269baa05a3b0c538/Formula/velero.rb), etc.) That being said, we have a few options:

- The `cmctl convert` command can be run without a cluster and converts between different versions of Kubernetes resources. This could be tested relatively easally with an input file and expected output file.
- We could mock the Kubernetes API server, this is what [kubeless](https://github.com/Homebrew/homebrew-core/blob/5b36ba740a61cab0a4f3b83c269baa05a3b0c538/Formula/kubeless.rb) seems to do. Although I suspect it would be a very brittle test and most Kubernetes tools in homebrew-core don't seem to take this approach, [only testing offline functionality where possible](https://github.com/Homebrew/homebrew-core/blob/5b36ba740a61cab0a4f3b83c269baa05a3b0c538/Formula/fluxctl.rb#L40-L42).

Let me know what you think is best, I'm happy to add the `cmctl convert` test if that would satasfy the requirement.

Reply:
I've added the `cmctl convert` test.

After doing some more reading into how other similar fomulas handle this I've also added a test that verifies that it fails in the expected way when trying to connect to a cluster. Based on this part of the formula cookbook:

> If the software cannot function without credentials or requires a virtual machine, docker instance, etc. to run, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected. This is preferred over mocking a dependency.